lib/vmapi-client.js

```diff
@@ -9,11 +9,120 @@
  */
 
 /*
- * vmapi-client.js
+ * This module contains the VmapiClient object which has functions for
+ * interaction with VMAPI.
+ *
  */
 
-var restify = require('restify');
+var restify = require('restify-clients');
+
+
+/*
+ * VMAPI has default values for some fields when the fields are not actually
+ * in the VM objects in Moray. For some of those fields such as:
+ *
+ *  autoboot: null
+ *  brand: null
+ *  firewall_enabled: false
+ *  nics: []
+ *  platform_build_stamp: null
+ *  resolvers: []
+ *  server_uuid: null
+ *  snapshots: []
+ *  state: null
+ *  tags: {}
+ *  zone_state: null
+ *  zonepath: null
+ *
+ * vmadm *also* always has these, so any time they're different we still want to
+ * do an update. The ones VMAPI always includes that are *not* always in vmadm
+ * objects are put into VMAPI_UNSET_FIELDS here so that we can detect their
```

Comment:
If I understand correctly, `VMAPI_UNSET_FIELDS` is a list of properties that are always set in VMAPI's responses? If that's the case the name of this variable seems to convey the opposite meaning. It seems it would be easier to understand if it was named something like `VMAPI_ALWAYS_SET_FIELDS`.